iiblib/iib_build_details_model.py

```diff
@@ -1,135 +1,285 @@
 class IIBBuildDetailsModel(object):
```

Comment:
I'm not gonna ask for all this to be reworked or anything, but I do have to wonder why we aren't using "attrs" library here.

- it's already used in other toolchain libraries, e.g. https://github.com/release-engineering/pushsource/blob/master/src/pushsource/_impl/model/ami.py
- it supports pretty much exactly what's attempted by these classes, including autogenerating \_\_eq\_\_ and similar methods (and I see manual implementation of these generated a fair bit of review comments).  Plus other things you might not think of, like sane implementations of \_\_hash\_\_, \_\_repr\_\_

For this kind of "plain old data" class, attrs library is giving loads of useful features and cutting down on boilerplate, so it's unfortunate we've got hand-rolled classes here instead.

Reply:
Thanks for pointing this out @rohanpm . I guess we did not used it because non of us knows about this library. I was searching for something but did not find anything good. I probably used wrong keywords for the search engine.

Reply:
Originally (when starting development of iiblib) I wanted to keep it simple, but I'm not against using attrs

Reply:
Do we want to create an enhancement ticket for changing it to attrs?

Reply:
Feel free to create it

Reply:
I created the ticket https://projects.engineering.redhat.com/browse/CLOUDDST-3412